engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/NewGameScreen.java

```diff
@@ -268,9 +268,7 @@ public void onOpened() {
         if (defaultGameplayModule != null) {
             gameplay.setSelection(defaultGameplayModule);
 
-            if (configDefaultModuleName.equalsIgnoreCase(DEFAULT_GAME_TEMPLATE_NAME)) {
-                setDefaultGeneratorOfGameplayModule(defaultGameplayModule);
-            }
+            setDefaultGeneratorOfGameplayModule(defaultGameplayModule);
```

Comment:
Why did this become necessary?
It looks like the default generator was only used under a specific condition. Now you are potentially overwriting the generator, regardless the value of `configDefaultModuleName`.
The changes to `DefaultWorldGeneratorPluginLibrary.java` are fixing the issue on their own btw.

Reply:
Yeah, that specific condition was "is the default module Josharias Survival?"

_why_ that specific condition, that is a good question, I wish I knew.

>  Now you are potentially overwriting the generator

**over**writing it? The problem I was encountering is that it wasn't written in the first place.

Under what conditions do we _not_ want the default generator to be set?

(I think some things here could be clearer if we separated some of this world-configuration code from the UI callbacks, but that's more than I wanted to bite off at the moment.)

Reply:
I was able to reproduce the necessity of this change.